code/modules/mob/living/living.dm

```diff
@@ -1395,3 +1397,39 @@
 	if(is_servant_of_ratvar(src))
 		eminence.selected_mob = src
 		to_chat(eminence, "<span class='brass'>You select [src].</span>")
+
+
+/mob/living/ghostize(can_reenter_corpse = TRUE, retain_sentience = FALSE )
+	..()
+	if(retain_sentience)
+		set_playable()
+
+/mob/living/attack_ghost(mob/user)
+	. = ..()
+	if(.)
+		return
+	give_mind(user)
+
+/mob/living/Topic(href, href_list)
+	if(href_list["activate"])
+		var/mob/dead/observer/ghost = usr
+		if(istype(ghost) && playable)
+			give_mind(ghost)
```

Comment:
Overrides /mob/Topic without calling ..()
(Needs this added)
```
if(..())
\treturn
```
Otherwise will break machine usage for mobs.